adoc/chapters/programming_interface.adoc

```diff
@@ -2936,845 +2933,841 @@ Any errors from the <<command-group>> after it has been submitted are handled by
 
-a@
-[source]
+'''
+
+.[apidef]#queue::get_info#
+[source,role=synopsis,id=api:queue-get-info]
 ----
-void wait_and_throw()
+template <typename Param>
+typename Param::return_type get_info() const
 ----
-a@ Performs a blocking wait for the completion of all enqueued tasks
-in the queue.  Synchronous errors will be reported through SYCL
-exceptions. Any unconsumed <<async-error,asynchronous errors>> will be passed to the
-<<async-handler>> associated with the queue or enclosing context.
-If no user defined [code]#async_handler# is associated with
-the queue or enclosing context, then an implementation-defined
-default <<async-handler>> is called to handle any errors, as
-described in <<subsubsec:exception.nohandler>>.
 
-a@
-[source]
+_Constraints:_ Available only when [code]#Param# is an information descriptor
+for the queue class.
+
+Each information descriptor specifies the return value and may also specify
+preconditions, exceptions that are thrown, etc.
+See <<sec:queue-info-descriptors>> for the queue information descriptors that
+are defined by the <<core-spec>>.
+
+'''
+
+.[apidef]#queue::get_backend_info#
+[source,role=synopsis,id=api:queue-get-backend-info]
 ----
-void throw_asynchronous()
+template <typename Param>
+typename Param::return_type get_backend_info() const
 ----
-a@ Checks to see if any unconsumed <<async-error,asynchronous errors>> have been produced by
-the queue and if so reports them by passing them to the
-<<async-handler>> associated with the queue or enclosing context.
-If no user defined [code]#async_handler# is associated with
-the queue or enclosing context, then an implementation-defined
-default <<async-handler>> is called to handle any errors, as
-described in <<subsubsec:exception.nohandler>>.
 
-a@
-[source]
+_Constraints:_ Available only when [code]#Param# is a backend information
+descriptor for the queue class.
+
+_Throws:_ An [code]#exception# with the [code]#errc::backend_mismatch# error
+code if the backend that corresponds with [code]#Param# is different from the
+backend that is associated with this queue.
+
+Each information descriptor specifies the return value and may also specify
+preconditions, additional exceptions that are thrown, etc.
+
+'''
+
+.[apidef]#queue::submit#
+[source,role=synopsis,id=api:queue-submit]
 ----
-template <typename Param> typename Param::return_type get_info() const
+template <typename T>
+event submit(T cgf)
 ----
-   a@ Queries this SYCL [code]#queue# for information requested by the
-      template parameter [code]#Param#.
-      The type alias [code]#Param::return_type# must be defined in
-      accordance with the info parameters in <<table.queue.info>> to
-      facilitate returning the type associated with the [code]#Param#
-      parameter.
 
-a@
-[source]
+_Effects:_ Submit a <<command-group-function-object>> to the queue, in order to
+be scheduled for execution on the device.
+
+_Returns:_ An event which the application can use to determine when the
```

Comment:
```
An event which represents the <<command>> that is submitted to the queue.
_note_ The even can be used to determine <<command>> associated with the <<command-group-function-object>> has completed.
```

And I guess we we should use `<<command>>` in al the `event` description.

Reply:
I know you already approved this, but I think your comment here is a good suggestion.  I updated the wording in 3cc87507efa5164df8c3395921b29b06fe40745f.

Reply:
I like it thanks!